src/openfermion/measurements/_quantum_subspace_expansion.py

```diff
@@ -0,0 +1,146 @@
+    https://arxiv.org/abs/1603.05681.
+    An extension to perform error mitigation based on QSE was proposed in
+    https://arxiv.org/abs/1807.10050.
+    If the expansion operators are symmetries or stabilizers
+    S-QSE is performed.
+
+    This function calculates the quantum subspace expansion of a given
+    Hamiltonian.
+    It is necessary to pass the expectation values of the Hamiltonian and the
+    additional operators.
+
+    Args:
+        hamiltonian(QubitOperator): Hamiltonian a QubitOperator.
+        expansion_operators(list, QubitOperator): List of QubitOperators
+        of the expansion.
+        expectation_values(QubitOperator): Expectation values as QubitOperator.
+
+    Returns:
+        spectrum_qse(list): Energy spectrum after QSE.
+
+    Raises:
+        TypeError: if hamiltonian is not a QubitOperator.
+        TypeError: if expansion_operators is not an array-like or
+                   QubitOperator.
+        TypeError: if expectation_values is not a QubitOperator.
+        ValueError: if length of expanded operator and length of expectation
+                    values is not equal.
+    """
+    if not isinstance(hamiltonian, QubitOperator):
+        raise TypeError('Hamiltonian must be a QubitOperator object.')
+    if not isinstance(expectation_values, QubitOperator):
+        raise TypeError('Hamiltonian must be a QubitOperator object.')
+    if not isinstance(expansion_operators, (QubitOperator, list,
+                                            numpy.ndarray)):
+        raise TypeError('List of stabilizers must be an array-like.')
+    # Write expansion operators as a list
+    if isinstance(expansion_operators, QubitOperator):
+        expansion_operators = list(expansion_operators)
+
+    # Check if the length of additional operator equals the length of
+    # expectation values.
+    additional_op = get_additional_operators(hamiltonian, expansion_operators)
+    if len(additional_op.terms) != len(expectation_values.terms):
+        raise ValueError('The number of Pauli strings do not match '
+                         'the number of expectation values.')
+
+    # Initialize matrices to store calculated values.
+    ham_mat = numpy.zeros((len(expansion_operators),
+                           len(expansion_operators)))
+    overlap_mat = numpy.zeros((len(expansion_operators),
+                               len(expansion_operators)))
+
+    for i in range(len(expansion_operators)):
+        for j in range(len(expansion_operators)):
+            op_to_trace = expectation_values * \
+                expansion_operators[j] * expansion_operators[i]
+            hop_to_trace = expectation_values * \
+                expansion_operators[j] * hamiltonian * expansion_operators[i]
+
+            # If an entry is 0 it will not be in the operators.
+            # An error will arise, hence we use try, expect to
+            # set the value in the matrix to 0.0
+            try:
```

Comment:
Is this the best way to access a coefficient of a term in an operator and return 0 otherwise, @kevinsung ?

Reply:
You can do `hop_to_trace.terms.get((), 0.0)`.